plugins/WebsiteMeasurable/MeasurableSettings.php

```diff
@@ -84,6 +85,10 @@ public function __construct(SitesManager\API $api, Plugin\Manager $pluginManager
 
     protected function init()
     {
+        if ($this->idMeasurableType !== Type::ID && $this->idMeasurableType !== MobileAppType::ID) {
+            return;
+        }
+
```

Comment:
My plan for this was to create classes for all properties and settings similar to https://github.com/piwik/piwik/blob/3.x-dev/plugins/WebsiteMeasurable/Settings/Urls.php and https://github.com/piwik/piwik/blob/3.0.0-rc1/plugins/WebsiteMeasurable/MeasurableSettings.php#L87-L88

So one `MeasurableSettings` should actually not define anything for another type and each plugin that defines a new type would do eg:

```php
        $this->urls = new Urls($this->idSite);
        $this->addSetting($this->urls);

        $this->excludedUrls = new ExcludedUrls($this->idSite);
        $this->addSetting($this->excludedUrls);
 ...
```

This is a bit of duplicated code and not best solution but most flexible since often it will be cases that types will want to use only some fields of it. 

Probably on top we will need to have something like your suggestion as well for cases where they actually want to have exact same fields, always. I will try to quickly define it like